src/main/java/com/example/msjira/controller/TeleSalesController.java

```diff
@@ -17,28 +17,32 @@ public class TeleSalesController {
     private final TeleSalesService teleSalesService;
 
     @GetMapping
-    public List<TeleSaleDto> getAllTeleSales(){
+    public List<TeleSaleDto> getAllTeleSales() {
         return teleSalesService.getAllTeleSales();
     }
 
     @GetMapping("/{teleSaleId}")
-    public TeleSaleDto getTeleSaleById(@PathVariable String teleSaleId){
+    public TeleSaleDto getTeleSaleById(@PathVariable String teleSaleId) {
         return teleSalesService.getTeleSaleById(teleSaleId);
     }
 
     @PostMapping
-    public void createTeleSale(@Valid @RequestBody TeleSaleReqDto teleSaleReqDto){
+    public void createTeleSale(@Valid @RequestBody TeleSaleReqDto teleSaleReqDto) {
         teleSalesService.createTeleSale(teleSaleReqDto);
     }
 
     @PutMapping("/{teleSaleId}")
-    public void updateTeleSale(@PathVariable String teleSaleId ,@Valid @RequestBody TeleSaleReqDto teleSaleReqDto){
-        teleSalesService.updateTeleSale(teleSaleId , teleSaleReqDto);
+    public void updateTeleSale(@PathVariable String teleSaleId, @Valid @RequestBody TeleSaleReqDto teleSaleReqDto) {
+        teleSalesService.updateTeleSale(teleSaleId, teleSaleReqDto);
     }
 
     @DeleteMapping("/{teleSaleId}")
-    public void deleteTeleSale(@PathVariable String teleSaleId){
+    public void deleteTeleSale(@PathVariable String teleSaleId) {
         teleSalesService.deleteTeleSale(teleSaleId);
     }
 
+    @PatchMapping("/{teleSaleId}/tasks/{taskId}/unassign")

```

Comment:
Bunu bir başa elə task controllerdən task/taskId/unassign patch atıb eləmək daha yaxşı olar. Çünki əgər telesale-in tasklarını görürsə əgər elə taskId-də deməli var əlində və ya reporter taskları görürsə elə taskId-də var həmçinin. Ona görə task controller-dən etsəh daha yaxşı olar.